src/plots/cartesian/axes.js

```diff
@@ -1203,6 +1210,51 @@ axes.calcTicks = function calcTicks(ax, opts) {
     return ticksOut;
 };
 
+function filterRangeBreaks(ax, ticksOut) {
+    if(ax.rangebreaks) {
+        // remove ticks falling inside rangebreaks
+        ticksOut = ticksOut.filter(function(d) {
+            return ax.maskBreaks(d.x) !== BADNUM;
+        });
+    }
+
+    return ticksOut;
+}
+
+function syncTicks(ax) {
+    // get the overlaying axis
+    var baseAxis = ax._mainAxis;
+
+    var ticksOut = [];
+    if(baseAxis._vals) {
+        for(var i = 0; i < baseAxis._vals.length; i++) {
+            // get the position of the every tick
+            var pos = baseAxis.l2p(baseAxis._vals[i].x);
+
+            // get the tick for the current axis based on position
+            var vali = ax.p2l(pos);
+            var val1 = ax.p2l(pos - 0.1);
+            var val2 = ax.p2l(pos + 0.1);
+            var d = 1 + Math.ceil(Math.log10(Math.abs(val2 - val1)));
```

Comment:
Although `Math.ceil` fixed some rounding issues, it looks like we should revert this part to have smooth ticks.
```suggestion
            var d = 1 + Math.round(Math.log10(Math.abs(val2 - val1)));
```

Reply:
Reverting... But now we are at the same point we were before when you suggest this change:

> We could try something like:
> 
> ```js
>             var vali = ax.p2l(pos);
>             var val1 = ax.p2l(pos - 0.5);
>             var val2 = ax.p2l(pos + 0.5);
>             var d = 1 + Math.round(Math.log10(Math.abs(val2 - val1)));
>             var e = Math.pow(10, -d);
>             var valR = Math.round(vali * e) / e;
>             var objR = axes.tickText(ax, valR);
>             var obj = axes.tickText(ax, vali);
>             obj.text = objR.text;
> ```